jerry-core/api/jerry.c

```diff
@@ -5513,6 +5514,185 @@ jerry_get_user_value (const jerry_value_t value) /**< jerry api value */
   return ecma_copy_value (CBC_SCRIPT_GET_USER_VALUE (script_p));
 } /* jerry_get_user_value */
 
+/**
+ * Returns a newly created source info structure corresponding to the passed script/module/function.
+ *
+ * @return a newly created source info, if at least one field is available, NULL otherwise
+ */
+jerry_source_info_t *
+jerry_get_source_info (const jerry_value_t value) /**< jerry api value */
+{
+  jerry_assert_api_available ();
+
+#if JERRY_FUNCTION_TO_STRING
+  jerry_source_info_t source_info;
+
+  source_info.enabled_fields = 0;
+  source_info.source_code = ECMA_VALUE_UNDEFINED;
+  source_info.function_arguments = ECMA_VALUE_UNDEFINED;
+  source_info.source_range_start = 0;
+  source_info.source_range_length = 0;
+
+  if (!ecma_is_value_object (value))
+  {
+    return NULL;
+  }
+
+  ecma_object_t *object_p = ecma_get_object_from_value (value);
+  cbc_script_t *script_p = NULL;
+
+  while (true)
+  {
+    switch (ecma_get_object_type (object_p))
+    {
+      case ECMA_OBJECT_TYPE_CLASS:
+      {
+        ecma_extended_object_t *ext_object_p = (ecma_extended_object_t *) object_p;
+        const ecma_compiled_code_t *bytecode_p = NULL;
+
+        if (ext_object_p->u.cls.type == ECMA_OBJECT_CLASS_SCRIPT)
+        {
+          bytecode_p = ECMA_GET_INTERNAL_VALUE_POINTER (ecma_compiled_code_t,
+                                                        ext_object_p->u.cls.u3.value);
+        }
+
+#if JERRY_MODULE_SYSTEM
+        if (ext_object_p->u.cls.type == ECMA_OBJECT_CLASS_MODULE)
+        {
+          ecma_module_t *module_p = (ecma_module_t *) object_p;
+
+          if (!(module_p->header.u.cls.u2.module_flags & ECMA_MODULE_IS_NATIVE))
+          {
+            bytecode_p = module_p->u.compiled_code_p;
+
+          }
+        }
+#endif /* JERRY_MODULE_SYSTEM */
+
+        if (bytecode_p == NULL)
+        {
+          return NULL;
+        }
+
+        ecma_value_t script_value = ((cbc_uint8_arguments_t *) bytecode_p)->script_value;
```

Comment:
This cast relies on the fact that the script_value is in the same place (same offset) in both cbc_{uint8,uint16}_arguments_t cases. Do we have an assert for this?

Reply:
This is a common technique these dfays, and afaik there is a check for it.

Reply:
I know it's common, and it is ok for me. I was just wandering if there is a check for it.